cmd/handler-utils.go

```diff
@@ -274,6 +274,18 @@ func httpTraceHdrs(f http.HandlerFunc) http.HandlerFunc {
 	return httptracer.TraceReqHandlerFunc(f, globalHTTPTraceFile, false)
 }
 
+// checkCertStatus returns an error when the server TLS certificate is revoked
+func checkCertStatus(f http.HandlerFunc) http.HandlerFunc {
+	return func(w http.ResponseWriter, r *http.Request) {
+		if globalIsCertRevoked {
+			writeErrorResponse(w, ErrRevokedCertificate, r.URL)
```

Comment:
Since you are using `checkCertStatus` for the browser as well it won't work since browser expects a JSON RPC response you will send an XML here, severely confusing the browser code. Can you write two handler funcs?  and move this to api-router.go and web-router.go

For web-router.go it is better to return an HTML not XML such that user gets to see the actual error upon loading the browser assets.